docs/src/UsingQAT/TargetingQIR.md

```diff
@@ -0,0 +1,248 @@
+settings are used to specify whether or not we require qubits and/or results to
+be present in the IR and whether we allow all QIS, or only those which were
+allowed listed:
+
+| Name             | Description                                      | Default value | CLI |
+| ---------------- | ------------------------------------------------ | ------------- | --- |
+| allowed-any-qis  | Whether or not to allow any quantum instruction. | true          | Yes |
+| requires-qubits  | Whether or not qubits are required in the IR.    | false         | Yes |
+| requires-results | Whether or not results are required in the IR.   | false         | Yes |
+| allowed-qis      | List of allowed QIS signatures                   |               | No  |
+
+Like the target profile validator, the `allowed-qis` is only available through
+the configuration file.
+
+## Adaptors
+
+In this section we discuss configuration of the available adaptors.
+
+### Adaptor deactivation - Used to disable or enable adaptors
+
+This configuration section allows the user to disable one or more adaptors. By
+default all adaptors are activated. While this does not mean that they actively
+perform any transformation of the IR, it does mean that the adaptor code is
+loaded and that the adaptor will be operating in accordance with its settings.
+
+The indiidual adaptors that can be disabled are listed here:
+
+| Name                                    | Description                               | Default value |
+| --------------------------------------- | ----------------------------------------- | ------------- |
+| disable-replacement-linking             | Disables Replacement linking.             | false         |
+| disable-llvm-optimization               | Disables LLVM optimizations.              | false         |
+| disable-remove-non-entrypoint-functions | Disables Remove Non-Entrypoint Functions. | false         |
+| disable-target-qis-mapping              | Disables Pass configuration.              | false         |
+| disable-target-profile-mapping          | Disables Post-transform optimisation.     | false         |
+| disable-straightline-code-requirement   | Disables Pre-transform validation.        | false         |
+| disable-static-resources                | Disables Static resource manipulation.    | false         |
+| disable-grouping                        | Disables Circuit separation.              | false         |
+
+Note that some of these adaptors are not configurable. One such example is the
+`remove-non-entrypoint-functions` which either is there or not. This particular
+adaptor works is just a traditional LLVM pass, but is delivered as an adaptor
+since QAT does not allow single passes to be added to the pipeline without an
+adaptor to manage its configuration. Moreover, since the pass does not require
+any configuration, its configuration section is empty.
+
+### LLVM optimizations - Enables specific LLVM optimizations before and after transformation
```

Comment:
Is it possible to configure qat such that built in llvm optimizations run in between custom qat transformations or does that require multiple qat invocations?

Reply:
No it is not. It is the assumption that any adaptor that may wish to run LLVM passes include those in itself with flags to tune the behaviour.